gematria/granite/graph_builder_model_inference.h

```diff
@@ -87,10 +88,22 @@ class GraphBuilderModelInference {
   // structure of a model based on the BasicBlockGraphBuilder class.
   GraphBuilderModelInference(
       std::unique_ptr<BasicBlockGraphBuilder> graph_builder,
-      const tflite::FlatBufferModel* tflite_model);
+      const tflite::FlatBufferModel* tflite_model,
+      std::unique_ptr<tflite::Interpreter> interpreter,
+      std::unique_ptr<std::vector<int>> input_tensor_to_idx,
```

Comment:
Why do we need this to be a unique pointer rather than just a vector? We do not allow `nullptr`, and `std::move()` of a `std::vector` is about as efficient as a move of a `std::unique_ptr<>`.